src/ray/common/client_connection.cc

```diff
@@ -84,7 +84,7 @@ ray::Status ServerConnection<T>::WriteMessage(int64_t type, int64_t length,
   boost::system::error_code error;
   WriteBuffer(message_buffers, error);
   if (error) {
-    return ray::Status::IOError(error.message());
```

Comment:
It looks strange why error.message() is corrupted. May we could print error.value() as well since that's valid?  That might also give some hints on what kind of errors encountered.

Reply:
Yeah, I think it'd be preferable to do `strerror(ec.value())` as in https://github.com/ray-project/ray/pull/2548 so that we have the error message, *even* despite the fact that this is probably not platform independent.

We can address this by factoring out all of the calls to `strerror(ec.value())` in a separate function.

Reply:
I am not certain that `ec.value()` is consistent with the errno numbers expected by `strerror`. There may be another way that's specific to boost. I am looking into it.